### ISSUE TITLE
Notebook requirements, configuration, and demos

### ISSUE THREAD

**Issue by jd-moulton** [amanzi/ats-short-course#2]
The Jupyter Notebooks are wonderful, but do have some limitations.   It would be great to tap into the expertise here to figure out a reasonable set of requirements, how to configure the notebook to support these requirements (i.e., do we need extensions and if so, which ones), and then include a demo notebook with simple examples covering these requirements.

For example, we could create a subdirectory to work on both the requirements and the demo notebook.  The requirements doc could be laid out as 

# Requirements

## Markdown cells
  * Math equations in isolation (standard)
  * Math equations with labels
  * ...

## Code cells
  * Need certain module for reason x
  * ...

**Comment by daniellivingston** [comment]
> Math equations in isolation (standard)
> Math equations with labels

IIRC, Jupyter should support MathJax by default, and it looks like we can configure some [MathJax parameters for equation labels/referencing](http://docs.mathjax.org/en/latest/input/tex/eqnumbers.html):

> The TeX input processing in MathJax can be configured to add equation numbers to displayed equations automatically. This functionality is turned off by default, but it is easy to configure MathJax to produce automatic equation numbers...
>
> It is also possible to set the tagging to 'all', so that every displayed equation will get a number, regardless of the environment used.
> You can use `\notag` or `\nonumber` to prevent individual equations from being numbered, and \tag{} can be used to override the usual equation number with your own symbol instead...
>
> You can use `\label` to give an equation an identifier that you can use to refer to it later, and then use `\ref` or `\eqref` within your document to insert the actual equation number at that location, as a reference.
